src/gg.regression.unity.bots/Editor/Scripts/RGLegacyInputUtility/RGLegacyInputInstrumentation.cs

```diff
@@ -0,0 +1,232 @@
+﻿#if UNITY_EDITOR
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Cecil.Pdb;
+using RegressionGames.RGLegacyInputUtility;
+using UnityEditor;
+using UnityEditor.Compilation;
+using UnityEngine;
+using Assembly = UnityEditor.Compilation.Assembly;
+
+namespace RegressionGames.Editor.RGLegacyInputUtility
+{
+    /*
+     * This class is responsible for hooking into the Unity build process and
+     * applying the wrapper input API instrumentation for the legacy input manager.
+     */
+    public class RGLegacyInputInstrumentation
+    {
+        [InitializeOnLoadMethod]
+        static void OnStartup()
+        {
+            // Safer to do any initial instrumentation within the editor update loop.
+            // Initial experiments with running directly from static initializers
+            // caused errors sometimes.
+            EditorApplication.update += SetUpHooks;
+        }
+
+        static void SetUpHooks()
+        {
+            if (EditorApplication.timeSinceStartup > 2.0)
+            {
+                EditorApplication.update -= SetUpHooks;
+                CompilationPipeline.assemblyCompilationFinished += OnAssemblyCompiled;
+                InstrumentExistingAssemblies();
+            }
+        }
+
+        private static bool IsAssemblyIgnored(string assemblyPath)
+        {
+            string fileName = Path.GetFileName(assemblyPath);
+            return fileName.Contains("RegressionGames")  // ignore RG assemblies
```

Comment:
We may want to put a developer focused README.md into the `Plugins` folder of gg.regression.unity.bots to remind people to update this file whenever they add a new package.